src/algorithms/channel/adapters/channel.cc

```diff
@@ -128,7 +132,7 @@ void Channel::connect(gr::top_block_sptr top_block)
 
     // Message ports
     top_block->msg_connect(nav_->get_left_block(), pmt::mp("telemetry_to_trk"), trk_->get_right_block(), pmt::mp("telemetry_to_trk"));
-    if ((trk_->get_right_block()->name() == "glonass_l1_ca_dll_pll_c_aid_tracking_cc") || (trk_->get_right_block()->name() == "glonass_l2_ca_dll_pll_c_aid_tracking_cc"))
+    if (dynamic_cast<glonass_dll_pll_c_aid_tracking_common *>(trk_->get_right_block().get()) && glonass_extend_correlation_ms_)
```

Comment:
On the other hand, if the connection is here because the tracking block adapter doesn't have access to the telemetry block, then you are working within the limitations of how the GNSS-SDR flowgraph is constructed.

In that case, I would consider querying the track block to see if it can accept a `"preamble_timestamp_samples"` message (see [`msg_ports_in()`](https://www.gnuradio.org/doc/doxygen/classgr_1_1basic__block.html#ae92faa0b6775de0100f6d202140403a2), which doesn't require type inspection at all.

In the end,
> Pragmatics must take precedence over elegance
For Nature is not impressed.

Do what you think is right. I'm just kibitzing from the sidelines

Reply:
Thanks. I'll try `msg_ports_in()` check as I can't decide where to put common ancestor file.